openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2897,132 +2904,233 @@ class GBSAParameterHandler(ParameterHandler):
+        else:
+            amber_cutoff = nonbonded_force.getCutoffDistance().value_in_unit(unit.nanometer)
+
+        if self.gb_model == 'OBC2':
+            gbsa_force = openmm_force_type()
+
+        else:
+            # We set these values in the constructor if we use the internal AMBER GBSA type wrapper
+            gbsa_force = openmm_force_type(solventDielectric=self.solvent_dielectric,
+                                           soluteDielectric=self.solute_dielectric,
+                                           SA=self.sa_model,
+                                           cutoff=amber_cutoff,
+                                           kappa=0,
+                                           )
+            # WARNING: If using a CustomAmberGBForce, the functional form is affected by whether
+            # the cutoff kwarg is None *during initialization*. So, if you initialize it with a
+            # non-None value, and then try to change it to None, you'll basically break things.
+
+        # Set the GBSAForce to have the same cutoff as NonbondedForce
+        #gbsa_force.setCutoffDistance(nonbonded_force.getCutoffDistance())
+        if amber_cutoff is not None:
+            gbsa_force.setCutoffDistance(amber_cutoff)
+
+        #gbsa_force.setNonbondedMethod(nonbonded_force.getNonbondedMethod())
+
+        if nonbonded_force.usesPeriodicBoundaryConditions():
+            # WARNING: The lines below aren't equivalent. The NonbondedForce and
+            # CustomGBForce NonbondedMethod enums have different meanings.
+            # More details:
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.NonbondedForce.html
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.GBSAOBCForce.html
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.CustomGBForce.html
+
+            #gbsa_force.setNonbondedMethod(simtk.openmm.NonbondedForce.CutoffPeriodic)
+            gbsa_force.setNonbondedMethod(simtk.openmm.CustomGBForce.CutoffPeriodic)
+        else:
+            #gbsa_force.setNonbondedMethod(simtk.openmm.NonbondedForce.NoCutoff)
+            gbsa_force.setNonbondedMethod(simtk.openmm.CustomGBForce.NoCutoff)
+
+        # Add all GBSA terms to the system. Note that this will have been done above
+        if self.gb_model == 'OBC2':
+            gbsa_force.setSolventDielectric(self.solvent_dielectric)
+            gbsa_force.setSoluteDielectric(self.solute_dielectric)
+            if self.sa_model is None:
+                gbsa_force.setSurfaceAreaEnergy(0)
+            else:
+                gbsa_force.setSurfaceAreaEnergy(self.surface_area_penalty)
+
+
+        # Iterate over all defined GBSA types, allowing later matches to override earlier ones.
+        atom_matches = self.find_matches(topology)
+
+        # Create all particles.
+
+        # !!! WARNING: CustomAmberGBForceBase expects different per-particle parameters
```

Comment:
I'd want @jchodera or @peastman to also eyeball this blurb.

Reply:
I think we should probably fork `customgbforces.py`, pull it from OpenMM into the openforcefield toolkit as an internal file, and modify it to behave as needed. `customgbforces` is essentially an internal OpenMM API and could be changed at any time, breaking our implementation.